src/App.tsx

```diff
@@ -14,11 +15,102 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  currentGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    currentGoods: ['Jam'],
+  };
+
+  selectOrRemove(inputGood: string) {
+    const newArrayOfGoods = [...this.state.currentGoods];
+
+    if (newArrayOfGoods.includes(inputGood)) {
+      const indexToDelete = newArrayOfGoods.indexOf(inputGood);
+
+      newArrayOfGoods.splice(indexToDelete, 1);
+    } else {
+      newArrayOfGoods.push(inputGood);
+    }
+
+    return newArrayOfGoods;
+  }
+
+  checkSelected(inputGood: string) {
+    return this.state.currentGoods.includes(inputGood);
+  }
+
+  render() {
+    const { currentGoods } = this.state;
+
+    return (
+      <div className="App">
+        <div className="goods">
+          <h1 className="goods__title">
+            Selected good:
+            {currentGoods.length !== 0 && (
+              <button
+                className="goods__clear btn btn-danger"
+                type="button"
+                onClick={() => {
+                  this.setState(() => ({
+                    currentGoods: [],
+                  }));
+                }}
+              >
+                Clear
+              </button>
+            )}
+          </h1>
+          <h2 className="goods__list">
+            {currentGoods.length === 0 && 'No goods selected'}
+            {currentGoods.length === 1 && `${currentGoods[0]} is selected`}
+            {currentGoods.length === 2 && `${currentGoods.join(' and ')} are selected`}
+            {currentGoods.length > 2 && `${currentGoods
+              .slice(0, -1)
+              .join(', ')} and ${currentGoods[currentGoods.length - 1]} are selected`}
```

Comment:
Will be better to create separate function that return correct string with selected goods

Reply:
thanks, done)